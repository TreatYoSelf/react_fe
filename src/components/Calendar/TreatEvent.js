import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function TreatEvent({ title, duration, time }) {
    return (
        <View>
            <Text style={styles.header}>{parseInt(time) < 12 ? `${time} AM` : `${time % 12} PM`}</Text>
            <View style={styles.container}>
                <Text style={styles.header}>{title}</Text>
                <Text style={styles.header}>{`${duration} Mins`}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'row',
        width: 300,
        padding: 10,
        backgroundColor: "#e6f7ff",
        borderWidth: 5,
        borderColor: '#003045',
        margin: 10
    },
    header: {
        fontWeight: 'bold',
        color: '#003045'
    },
    selected: {
        backgroundColor: "blue",
    }
});