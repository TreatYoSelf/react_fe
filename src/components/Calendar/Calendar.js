import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View, SectionList } from 'react-native';
import TreatEvent from './TreatEvent';
// import { gql } from 'apollo-boost';
// import { useQuery } from '@apollo/react-hooks';
import { mockEvents } from '../../mockData/mockTreat';

export default function Calendar() {
    //do calcs here to group into a single date object 
    //dont need to parse float just JSON parse on fetch 

    const calendarEvents = mockEvents.reduce((eventByDay, event) => {
        let {eventName, eventStartTime, eventEndTime} = event;
        eventStartTime = new Date(parseFloat(eventStartTime));
        eventEndTime = new Date(parseFloat(eventEndTime));
        const duration = Math.floor((eventEndTime.getTime() - eventStartTime.getTime()) / 60000);
        const eventTime = eventStartTime.getHours();
        const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
        const eventDay = days[eventStartTime.getDay()];
        const eventData = {
            eventName,
            duration,
            eventTime
        }

        if (!eventByDay[eventDay]) {
            eventByDay[eventDay] = [eventData]  
        } else {
            eventByDay[eventDay].push(eventData)
        }

        return eventByDay;
    }, {})

    //do a map here if object.key has values, create a section obj with ID, title and array that contains all values for that key
    const eventsByDay = Object.keys(calendarEvents).map((event, index) => {
        return {
            id: index,
            title: event,
            data: calendarEvents[event]
        }
    })

    //store events by weekday
    //create a sectionList for those events with the section being the day if available
    //use above to create section list 
    //http://www.reactnativeexpress.com/sectionlist


    //add border bottom to text for hrule 
    //map over events

    return (
        <View style={styles.container}>
            <Text style={styles.header}>Your Treats This Week</Text>
            <SectionList
                sections={eventsByDay}
                keyExtractor={(item, index) => item + index}
                renderItem={({ item }) => <TreatEvent title={item.eventName} duration={item.duration}/>}
                renderSectionHeader={({ section: { title } }) => (
                    <Text style={styles.day}>{title}</Text>
                )}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        // flex:1,
        alignItems: "center",
        justifyContent: "space-around"
        // flexDirection: "column"
        // height: "100%",
        // width: 100
    },
    header: {
        fontSize: 27,
        fontWeight: 'bold'
    },
    day: {
        fontSize: 25,
        marginTop: 30
    },
    categories: {
        backgroundColor: "#fff",
        flexGrow: 0,
        // height: 60
        // width: 100
    }
})
