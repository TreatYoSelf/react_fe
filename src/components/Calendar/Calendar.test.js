import React from 'react';
import renderer, { act } from 'react-test-renderer';

import Calendar from './Calendar';

describe('<Calendar />', () => {
    it('renders correctly', async () => {
        let tree;
        tree = renderer.create(<Calendar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});