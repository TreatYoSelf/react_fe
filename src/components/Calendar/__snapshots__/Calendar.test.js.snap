// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "space-around",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#003045",
        "fontSize": 27,
        "fontWeight": "bold",
      }
    }
  >
    Your Treats This Week
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "duration": 50,
              "eventName": "Hike",
              "eventTime": 12,
            },
          ],
          "id": 0,
          "title": "Wednesday",
        },
        Object {
          "data": Array [
            Object {
              "duration": 66,
              "eventName": "Spa Day",
              "eventTime": 0,
            },
            Object {
              "duration": 63,
              "eventName": "Yoga",
              "eventTime": 0,
            },
          ],
          "id": 1,
          "title": "Saturday",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "duration": 50,
              "eventName": "Hike",
              "eventTime": 12,
            },
          ],
          "id": 0,
          "title": "Wednesday",
        },
        Object {
          "data": Array [
            Object {
              "duration": 66,
              "eventName": "Spa Day",
              "eventTime": 0,
            },
            Object {
              "duration": 63,
              "eventName": "Yoga",
              "eventTime": 0,
            },
          ],
          "id": 1,
          "title": "Saturday",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
        3,
      ]
    }
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "marginTop": 30,
            }
          }
        >
          Wednesday
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "honeydew",
                "borderWidth": 5,
                "flexDirection": "row",
                "justifyContent": "space-around",
                "margin": 10,
                "padding": 10,
                "width": 300,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Hike
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            50 Mins
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 25,
              "marginTop": 30,
            }
          }
        >
          Saturday
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "honeydew",
                "borderWidth": 5,
                "flexDirection": "row",
                "justifyContent": "space-around",
                "margin": 10,
                "padding": 10,
                "width": 300,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Spa Day
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            66 Mins
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "honeydew",
                "borderWidth": 5,
                "flexDirection": "row",
                "justifyContent": "space-around",
                "margin": 10,
                "padding": 10,
                "width": 300,
              },
            ]
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 50,
                "width": 50,
              }
            }
          />
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Yoga
          </Text>
          <Text
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            63 Mins
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
